### PR TITLE
Add optional notification warning in the upload view

### LINKED ISSUE #4233: Add warnings in devhub about submission restrictions for legacy add-ons
Per https://blog.mozilla.org/addons/2016/11/23/add-ons-in-2017/ we'll be restricting *new* submissions of legacy add-ons to AMO when Firefox 53 is released (by end of Q1).

The add-on submission flows (new add-on, new version, new file) need to inform developers of the upcoming restrictions before that date. Per [the PRD](https://docs.google.com/document/d/1BdiJSYD4aOMClasGxx5wVShHEk8VbxefE3Gf_6jpgkQ/edit#) this should follow the UX proposed in #3634

### PR DESCRIPTION
**Pull request by diox**
Fix #4233

This is set by admins through the Config model

### CODE REVIEW

File: src/olympia/devhub/templates/devhub/addons/submit/base.html [hunk 1]
```diff
@@ -13,6 +13,12 @@
 {% endblock %}
 
 {% block content %}
+{% if submit_notification_warning %}
+    <div class="notification-box warning">
+        {# The submit notification warning is controlled by admins, HTML is allowed #}
+        {{ submit_notification_warning|safe }}
+    </div>
+{% endif %}
```
Comment (EnTeQuAk) [inline src/olympia/devhub/templates/devhub/addons/submit/base.html]: I feel like this requires a test that at least the box is visible and can be extended to verify that html is indeed allowed. Though I didn't look into if there are any tests already covering that.

Comment (diox) [inline src/olympia/devhub/templates/devhub/addons/submit/base.html]: Yeah, I wasn't sure it really deserved a test, but I can add a simple one testing that HTML works.

Comment (diox) [inline src/olympia/devhub/templates/devhub/addons/submit/base.html]: Test added.

Comment (EnTeQuAk) [inline src/olympia/devhub/templates/devhub/addons/submit/base.html]: perfect, thanks!

**Pull request merged**